src/sfdc48/salesforce_bulk_endpoint.bal

```diff
@@ -0,0 +1,369 @@
+        string path = prepareUrl([SERVICES, ASYNC, BULK_API_VERSION, JOB, jobId]);
+        http:Request req = new;
+        var response = self.httpClient->get(path, req);
+        if(JSON == jobDataType) {
+            json|ConnectorError jobResponse = checkJsonPayloadAndSetErrors(response);
+            if (jobResponse is json){            
+                JobInfo jobInfo = check JobInfo.constructFrom(jobResponse);
+                return jobInfo;
+            } else {
+                return jobResponse;
+            }
+        } else {
+            xml|ConnectorError jobResponse = checkXmlPayloadAndSetErrors(response);
+            if (jobResponse is xml){            
+                JobInfo jobInfo = check createJobRecordFromXml(jobResponse);
+                return jobInfo;
+            } else {
+                return jobResponse;
+            }
+        }
+        
+    }
+
+    # Close a job
+    #
+    # + bulkJob - job to be closed 
+    # + return - job info after the state change of the job
+    public remote function closeJob(BulkJob bulkJob) returns @tainted error|JobInfo {
+        string jobId = bulkJob.jobId;
+        string path = prepareUrl([SERVICES, ASYNC, BULK_API_VERSION, JOB, jobId]);
+        http:Request req = new;
+        req.setJsonPayload(JSON_STATE_CLOSED_PAYLOAD);
+        var response = self.httpClient->post(path, req);
+        json|ConnectorError jobResponse = checkJsonPayloadAndSetErrors(response);
+        if(jobResponse is json){
+            JobInfo jobInfo = check JobInfo.constructFrom(jobResponse);
+            return jobInfo;
+        } else {
+            return jobResponse;
+        }
+    }
+
+    # abort a job
+    #
+    # + bulkJob - job to be aborted 
+    # + return - job info after the state change of the job
+    public remote function abortJob(BulkJob bulkJob) returns @tainted error|JobInfo {
+        string jobId = bulkJob.jobId;
+        string path = prepareUrl([SERVICES, ASYNC, BULK_API_VERSION, JOB, jobId]);
+        http:Request req = new;
+        req.setJsonPayload(JSON_STATE_CLOSED_PAYLOAD);
+        var response = self.httpClient->post(path, req);
+        json|ConnectorError jobResponse = checkJsonPayloadAndSetErrors(response);
+        if(jobResponse is json){
+            JobInfo jobInfo = check JobInfo.constructFrom(jobResponse);
+            return jobInfo;
+        } else {
+            return jobResponse;
+        }
+    }
+};
+
+
+# Job object
+public type BulkJob object {
+    string jobId;
+    string jobDataType;
+    http:Client httpClient;
+
+    public function __init(string jobId, string jobDataType, http:Client httpClient) {
+        self.jobId = jobId;
+        self.jobDataType = jobDataType;
+        self.httpClient = httpClient;
+    }
+
+    # Add batch to the job
+    #
+    # + content - batch content 
+    # + return - batch info or error
+    public function addBatch(json|string|xml|io:ReadableByteChannel content) returns @tainted error|BatchInfo{
```

Comment:
We need to make this as remote function as we are doing remote calls

Reply:
BulkJob is not of client type. It's just an object. If we make it a client object, it'll be conflicting with our design right?

Reply:
I think we need to make BulkJob as a client object since we are doing remote calls. So it will be BasicClient -> BulkClient -> BulkJob

Reply:
Done